### ISSUE TITLE
Multi-Band MelGAN v2 finetuning LJSpeech turns pure noise as soon as discriminator turns on

### ISSUE THREAD

**Issue by ZDisket** [kan-bayashi/ParallelWaveGAN#171]
No matter what I do, when finetuning any of my smaller datasets using the LJSpeech Multi-Band MelGAN v2 model as pretrained and the same config, as soon as the discriminator turns on (I train generator only until about 24k steps, works with PWGAN) the speech becomes pure noise. Prior to that, the audio samples have noise. I don't have this issue with regular ParallelWaveGAN. Any ideas? 

The attached file contains some predictions.
[Predictions.zip](https://github.com/kan-bayashi/ParallelWaveGAN/files/4834545/Predictions.zip)

**Comment by dathudeptrai** [comment]
@ZDisket there are many ways you can try

- MultiBand Parallelwavegan.
- MultiBand MelGAN with Parallelwavegan discriminator.
- Training from scratch with original setting exactly, 24k steps is not enough for G of Mb-melgan stable.

**Comment by kan-bayashi** [comment]
This is interesting.
I have a similar experience using PWG, but not yet tried with MB-MelGAN.
How about the following ideas?

- Use a fixed discriminator
- Use small LR settings, e.g., default LR after 600k iters

**Comment by ZDisket** [comment]
@dathudeptrai @kan-bayashi 
I've tried starting training with the discriminator on at 0 steps, meaning no steps with generator only. The audio samples are bad but they aren't pure noise and seem to be making progress 10k steps in, so I'll continue and if it works close this issue.

**Comment by kan-bayashi** [comment]
@ZDisket How was the results?

**Comment by ZDisket** [comment]
@kan-bayashi Sorry, I got sidetracked on other things like FastSpeech2. I'll close and reopen if I still have a problem.

**Comment by ZDisket** [comment]
I just got back into this. Training mb-melgan with discriminator on since 0 steps works to not make pure noise, but it never sounded good. In the end, I combined mb-melgan generator with pwgan discriminator (in TensorflowTTS implementation), slashed the learning rates, and the result was pretty good.

**Comment by kan-bayashi** [comment]
@ZDisket Thank you for sharing your insight!
You mean the use of pretrained PWG discriminator?

**Comment by ZDisket** [comment]
@kan-bayashi No, multi-band melgan generator load from weights, but train new pwgan discriminator from scratch. Performs better than pwgan and mb-melgan in my datasets.

**Comment by kan-bayashi** [comment]
That is interesting. I will check it!

**Comment by ZDisket** [comment]
@kan-bayashi Sounds good. Let me know how it goes for you, I'm interested.

**Comment by LqNoob** [comment]
@ZDisket 
1. Using multi-band melgan generator + pwgan discriminator to train the model directly, what is the final synthesized sound quality under the verification set? I didn’t get good sound quality；
[predictions.zip](https://github.com/kan-bayashi/ParallelWaveGAN/files/5370068/predictions.zip)
2. What does your tensorboard look like? Can you give the result of tensorboard？ Following is my training process：
![image](https://user-images.githubusercontent.com/28683383/95838040-e471de80-0d73-11eb-9f54-49e0d744ea7d.png)
![image](https://user-images.githubusercontent.com/28683383/95838113-f81d4500-0d73-11eb-952e-9640f48c3402.png)

**Comment by kan-bayashi** [comment]
@LqNoob 
> Using multi-band melgan generator + pwgan discriminator to train the model directly,

You mean the training from scratch? If so, that makes no sense.
MB-MelGAN + PWG discriminator works well for the adaptation case.
It is better to use the MelGAN discriminator for the training from scratch.

**Comment by LqNoob** [comment]
@kan-bayashi  Thanks for your response.

> MB-MelGAN + PWG discriminator works well for the adaptation case.

Now I am training the model in this way. How many steps can loss converge? How is the sound quality generated by the this model? If you have already trained the model, can you give the result of tensorboard?
